ONSdigital/sdc-int-rh-service
# 3400 Local firestore emulator

# Motivation and Context

We want to be able to build, run and test the service locally without depending on cloud environments.

# What has changed



- Consolidate the docker compose bits into one docker compose file
- Add firestore emulator to docker compose
- Change docker compose script so that it defaults to local, but can still be configured to target a cloud firestore
- Add Makefile with some useful shortcuts

# How to test?



Ensure you don't have `GOOGLE_CLOUD_PROJECT` set in your environment and run `make up`, and check the RH services logs with `docker logs -f rh-service`. You should see it start up successfully against project `dummy-local`, indicating it is successfully using the local firestore emulator by default. You should also still be able to to set `GOOGLE_CLOUD_PROJECT` and `DOCKER_GCP_CREDENTIALS` as guided in the README to target a real cloud firestore, optionally.

# Links



https://trello.com/c/J3xNjIeD/3400-investigate-running-testing-rh-fully-locally
